boa3_test/examples/htlc.py

```diff
@@ -134,11 +134,13 @@ def onNEP17Payment(from_address: UInt160, amount: int, data: Any):
     :raise AssertionError: raised if `from_address` length is not 20
     """
     # the parameters from and to should be 20-byte addresses. If not, this method should throw an exception.
-    if from_address is not None:
+    aux_var = from_address is not None  # TODO: remove this variable when type checking the values after the if is fixed
+    if aux_var:
         assert len(from_address) == 20
 
     # this validation will verify if Neo is trying to mint GAS to this smart contract
-    if from_address is None and runtime.calling_script_hash == GAS_SCRIPT:
+    aux_var = from_address is None      # TODO: remove this variable when type checking the values after the if is fixed
+    if aux_var and runtime.calling_script_hash == GAS_SCRIPT:
```

Comment:
There's any change in the behavior between these two codes?

Reply:
Yes, I wrote in mode detail at **(Optional) Additional context**, but trying to use any of the following: `if from_address is not None:`, `if from_address is None and runtime.calling_script_hash == GAS_SCRIPT:`, `if not isinstance(from_address, None):` and `if isinstance(from_address, None) and runtime.calling_script_hash == GAS_SCRIPT:` will result in a compilation error, because it will assume `from_address` is `None`. I tried using `isinstance` on the `development` branch and it also didn't work

Reply:
Also there is a HTLC example in the docs as well, can you change the code there too?